hibernate-core/src/main/java/org/hibernate/bytecode/internal/bytebuddy/ByteBuddyState.java

```diff
@@ -186,13 +186,25 @@ void clearState() {
 
 	private Class<?> load(Class<?> referenceClass, TypeCache<TypeCache.SimpleKey> cache,
 			TypeCache.SimpleKey cacheKey, Function<ByteBuddy, DynamicType.Builder<?>> makeProxyFunction) {
-		return cache.findOrInsert(
-				referenceClass.getClassLoader(),
-				cacheKey,
-				() -> make( makeProxyFunction.apply( byteBuddy ) )
-						.load( referenceClass.getClassLoader(), resolveClassLoadingStrategy( referenceClass ) )
-						.getLoaded(),
-				cache );
+
+		PrivilegedAction<Class<?>> delegateToPrivilegedAction =
+				new PrivilegedAction<Class<?>>() {
+
+					@Override
+					public Class<?> run() {
+						return cache.findOrInsert(
+								referenceClass.getClassLoader(),
```

Comment:
just wondering, but the privileged action didn't necessary need to wrap the "find" operation on the cache, right?

If I'm understanding correctly what needs to be wrapped is just the "make( ..)" method ?

Reply:
You're right, but I didn't to nitpick here since I thought this was only called during boot. Are you saying this is also called during runtime?

Reply:
It's ok, don't mean to delay the patch about this.
And yes it's only a one-off during bootstrap, but on a non-trivial model we might be spending minutes on enhancement processing, in particular this cache is on the hot area.

We can polish it further later, but it's useful to have confirmation that it's the `make` that needs to be wrapped only.

Reply:
+1 for reducing the privileged action to a smaller amount of code.

Reply:
Failure with the https://github.com/scottmarlow/hibernate-orm/commit/6c478f9d6fb9f6ab47519c5ac196f6b8be086e98 is:
> [javatest.batch] ^[[0m^[[0m14:50:58,715 INFO  [stdout] (Thread-196) Caused by: java.lang.IllegalStateException: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "defineClass")" in code source "(vfs:/content/jpa_core_entityManager_vehicles.ear/jpa_core_entityManager_appmanaged_vehicle_ejb.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.jpa_core_entityManager_vehicles.ear.jpa_core_entityManager_appmanaged_vehicle_ejb.jar" from Service Module Loader")
> [javatest.batch] ^[[0m^[[0m14:50:58,715 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.dynamic.loading.ClassInjector$UsingLookup.injectRaw(ClassInjector.java:1640)
> [javatest.batch] ^[[0m^[[0m14:50:58,715 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.dynamic.loading.ClassInjector$AbstractBase.inject(ClassInjector.java:118)
> [javatest.batch] ^[[0m^[[0m14:50:58,715 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.dynamic.loading.ClassLoadingStrategy$UsingLookup.load(ClassLoadingStrategy.java:519)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:101)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:6317)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at org.hibernate@6.1.6-SNAPSHOT//org.hibernate.bytecode.internal.bytebuddy.ByteBuddyState.lambda$load$0(ByteBuddyState.java:193)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:168)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     ... 116 more
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196) Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "defineClass")" in code source "(vfs:/content/jpa_core_entityManager_vehicles.ear/jpa_core_entityManager_appmanaged_vehicle_ejb.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.jpa_core_entityManager_vehicles.ear.jpa_core_entityManager_appmanaged_vehicle_ejb.jar" from Service Module Loader")
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at org.wildfly.security.elytron-base@2.0.0.Final//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:309)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at org.wildfly.security.elytron-base@2.0.0.Final//org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:201)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at java.base/java.lang.invoke.MethodHandles$Lookup.defineClass(MethodHandles.java:926)
> [javatest.batch] ^[[0m^[[0m14:50:58,716 INFO  [stdout] (Thread-196)     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at java.base/java.lang.reflect.Method.invoke(Method.java:566)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at net.bytebuddy.utility.Invoker$Dispatcher.invoke(Unknown Source)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.utility.dispatcher.JavaDispatcher$Dispatcher$ForNonStaticMethod.invoke(JavaDispatcher.java:1032)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.utility.dispatcher.JavaDispatcher$ProxiedInvocationHandler.invoke(JavaDispatcher.java:1162)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at net.bytebuddy//com.sun.proxy.$Proxy82.defineClass(Unknown Source)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     at net.bytebuddy//net.bytebuddy.dynamic.loading.ClassInjector$UsingLookup.injectRaw(ClassInjector.java:1638)
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)     ... 122 more
> [javatest.batch] ^[[0m^[[0m14:50:58,717 INFO  [stdout] (Thread-196)

Reply:
It's actually the load call that needs to be wrapped